folio-org/stripes-components
# STCOM-497: Updates for <RepeatableField>

In this PR I have made some refactoring and updates for the `<RepeatableField>`-component.

**Ticket:** https://issues.folio.org/browse/STCOM-497

## Changes
- Added `id` prop and prefixed ID for the "add new"-button
- Made `onRemove` optional
- Updated style for content div to ensure that the content is rendered in full width
- Changed so that the `legend`-prop always renders a `<Headline>`
- Updated "emptyMessage" to render the `<EmptyMessage>`-component
- Minor change for `<EmptyMessage>`-component to allow for adding custom classes
- Added examples for multiple fields and how to add a custom remove button
- Updated the readme with updated props and more usage examples
- Added a few more tests

## Redux Form / React Final Form
I tested out the component using both Redux Form and React Final Form and the component API for the `<FieldArray>`-component is very similar (if not identical) so I changed the readme to reflect that.

Previously, the `formFieldArray` helper function automatically added the `onRemove` callback and this is still the default. 

In addition to this, I changed it so that it's possible to overwrite this behavior by passing a custom `onRemove` callback that will receive `fields` and `index` as parameters. It's also possible to set `onRemove={false}` to hide the remove buttons all together when using redux form/final form.

## Tested out in ui-inventory
I did some testing to see if my changes would work when replacing the local `RepeatableField`-component in ui-inventory and it seems to work as intended.

These changes can be tested on the `test-stripes-components-repeatablefield` branch in ui-inventory. I updated all instances of `<RepeatableField>` in `ItemForm.js`. 

To test in FOLIO:
1. Click on a result in Inventory
2. Scroll down and click on an item
3. Click edit item
